src/actions/channels.js

```diff
@@ -711,14 +714,20 @@ export function viewChannel(channelId: string, prevChannelId: string = ''): Acti
         const {myMembers} = getState().entities.channels;
         const member = myMembers[channelId];
         if (member) {
+            if (isManuallyUnread(channelId, getState())) {
+                actions.push({
+                    type: ChannelTypes.REMOVE_MANUALLY_UNREAD,
+                    data: {channelId},
+                });
+            }
             actions.push({
                 type: ChannelTypes.RECEIVED_MY_CHANNEL_MEMBER,
                 data: {...member, last_viewed_at: new Date().getTime()},
             });
             dispatch(loadRolesIfNeeded(member.roles.split(' ')));
         }
 
-        const prevMember = myMembers[prevChannelId];
+        const prevMember = myMembers[preChanId];
         if (prevMember) {
```

Comment:
Right now, `myMembers[preChanId]` will just be undefined if the previous channel was manually marked as unread, but that's difficult to see without following the code closely. I'd suggest that we can just be explicit about this like you're doing in `markChannelAsViewed`
```javascript
const prevMember = myMembers[prevChannelId];
if (prevMember && !prevChannelManuallyUnread) {
```